.github/workflows/sycl_linux_build_and_test.yml

```diff
@@ -110,30 +114,30 @@ jobs:
       run: cmake --build $GITHUB_WORKSPACE/build
     # TODO allow to optionally disable in-tree checks
     - name: check-llvm
-      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
+      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' && inputs.only_test_e2e_changed == 'false' }}
```

Comment:
I wanted to leave it for a subsequent PR(s) because different projects need different logic. For example clang should track both `clang/` and `llvm/` and possibly other directories.

Reply:
> I wanted to leave it for a subsequent PR(s) because different projects need different logic.

The logic is the same, but implementation is more complex. The implementation complexity increases with the amount of project dependencies located outside of projects main source directory. 
Anyway, I suggest doing right away. It brings more value than the proposed patch (it will save time on testing the compiler for the patches changing DPC++ runtime only) which will be thrown away by subsequent PR(s).